scripts/test-alpn-negotiation.py

```diff
@@ -360,7 +359,150 @@ def main():
     node.next_sibling = ExpectClose()
     conversations["renegotiation 2nd handshake alpn"] = conversation
 
+    # resumption without alpn change
+
+    conversation = Connect(host, port)
+    node = conversation
+    ciphers = [CipherSuite.TLS_RSA_WITH_AES_128_CBC_SHA]
+    ext = {ExtensionType.alpn: ALPNExtension().create([bytearray(b'http/1.1')]),
+	   ExtensionType.renegotiation_info:None}
+    node = node.add_child(ClientHelloGenerator(ciphers, extensions=ext))
+    ext = {ExtensionType.renegotiation_info: None,
+           ExtensionType.alpn: lambda _, ext: ext == ALPNExtension().create([bytearray(b'http/1.1')])}
+    node = node.add_child(ExpectServerHello(extensions=ext))
+    node = node.add_child(ExpectCertificate())
+    node = node.add_child(ExpectServerHelloDone())
+    node = node.add_child(ClientKeyExchangeGenerator())
+    node = node.add_child(ChangeCipherSpecGenerator())
+    node = node.add_child(FinishedGenerator())
+    node = node.add_child(ExpectChangeCipherSpec())
+    node = node.add_child(ExpectFinished())
+    node = node.add_child(AlertGenerator(AlertLevel.warning,
+                                         AlertDescription.close_notify))
+    node = node.add_child(ExpectAlert())
+    close = ExpectClose()
+    node.next_sibling = close
+    node = node.add_child(ExpectClose())
+    node = node.add_child(Close())
+    node = node.add_child(Connect(host, port))
+    close.add_child(node)
+
+    node = node.add_child(ResetHandshakeHashes())
+    node = node.add_child(ResetRenegotiationInfo())
+    ext = {ExtensionType.alpn: ALPNExtension().create([bytearray(b'http/1.1')]),
+	   ExtensionType.renegotiation_info:None}
+    node = node.add_child(ClientHelloGenerator(ciphers, extensions=ext))
+    ext = {ExtensionType.renegotiation_info:None,
+           ExtensionType.alpn: lambda _, ext: ext == ALPNExtension().create([bytearray(b'http/1.1')])}
```

Comment:
no need to use lambdas any more if you don't use the session information

`ExtensionType.alpn: ALPNExtension().create([bytearray(b'http/1.1')])` will have exact same effect